go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -182,10 +183,20 @@ func (oc *DefaultNetworkController) syncEgressFirewall(egressFirewalls []interfa
 
 	// delete acls from all switches, they reside on the port group now
 	if len(egressFirewallACLs) != 0 {
-		err = libovsdbops.RemoveACLsFromLogicalSwitchesWithPredicate(oc.nbClient, func(item *nbdb.LogicalSwitch) bool { return true },
-			egressFirewallACLs...)
+		// optimize the predicate to exclude switches that don't reference deleting acls.
+		// it is safe to do, since this function transacts the ops.
+		aclsToDelete := sets.NewString()
+		for _, acl := range egressFirewallACLs {
+			aclsToDelete.Insert(acl.UUID)
+		}
```

Comment:
Two nits here:

- The ACLs are already iterated above so perhaps we can use that iteration to get the UUIDs out of them? But...
- I see how this works, but I am missing why it would not be better to consider only ACLs of each batch, so defining aclsToDelete within the batch function I mean.

Reply:
no reason really, we can do that per-batch